### PR TITLE
1.Windows compatibility. 2. Conflicts merged. 3.All discv4 tests

### PR DESCRIPTION
**Pull request by FrankSzendzielarz**

### CODE REVIEW

**Comment by fjl** [comment]
Looks OK to me, but please remove `.vscode`

**Comment by FrankSzendzielarz** [comment]
I left it in deliberately and mentioned it in the readme.md. Maybe cleaner would be to remove it and add the launch.json to the readme.

**Comment by FrankSzendzielarz** [comment]
This update to the pull request includes, in addition to the original windows compatibility:
1. Upstream merge conflicts resolved.
2. DevP2P client docker images, that include the environment variables for a bootnode to enable discovery
3. Validator docker image for testing the client for v4 ping
4. Go test project that can be run standalone or as part of 3) 
5. Client images can now include an "enode.sh" that can offer a client specific way of getting the enode target id
6. If not supplied in 5., the devp2p test in 4. can determine the enode based on ec recovery of Ping response
This sets the basic framework for development of the remainder of the tests.

File: clients/ethereumj_master/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Docker container spec for building the master branch of EthereumJ.
```
Comment (holiman) [inline clients/ethereumj_master/Dockerfile]: This file has been removed, don't add it back

File: clients/ethereumj_master/harmony.sh [hunk 2]
```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
```
Comment (holiman) [inline clients/ethereumj_master/harmony.sh]: This is also removed

**Comment by FrankSzendzielarz** [comment]
@fjl @holiman 
1. Got rid of those extra file you wanted removing @holiman 
2. @fjl All the tests for discv4 are now in. They all pass, but will spend time tomorrow making sure they do what they are supposed to do.

File: clients/go-ethereum_master/geth.sh [hunk 3]
```diff
@@ -123,4 +123,4 @@ fi
 
 # Run the go-ethereum implementation with the requested flags
 echo "Running go-ethereum..."
-/geth $FLAGS --nat=none --rpc --rpcaddr "0.0.0.0" --rpcapi "admin,debug,eth,miner,net,personal,shh,txpool,web3" --ws --wsaddr "0.0.0.0" --wsapi "admin,debug,eth,miner,net,personal,shh,txpool,web3" --wsorigins "*"
+/geth --verbosity 6 $FLAGS --nat=none --rpc --rpcaddr "0.0.0.0" --rpcapi "admin,debug,eth,miner,net,personal,shh,txpool,web3" --ws --wsaddr "0.0.0.0" --wsapi "admin,debug,eth,miner,net,personal,shh,txpool,web3" --wsorigins "*"
```
Comment (holiman) [inline clients/go-ethereum_master/geth.sh]: please don't -- `4` is `DEBUG`, `5` is `TRACE`, `6` means is everything and their grandmothers. The logs will be humoungous if we do this.

**Pull request merged**